src/n-acd-probe.c

```diff
@@ -236,9 +236,9 @@ static int n_acd_probe_link(NAcdProbe *probe) {
         }
 
         /*
-         * Add the ip address to the map, if it is not already there.
+         * Add the ip address to the map, it is not already there.
          */
-        if (n_acd_probe_is_unique(probe)) {
+        if (probe->acd->fd_bpf_map != -1 && n_acd_probe_is_unique(probe)) {
```

Comment:
This probably should be changed as suggested in https://github.com/nettools/n-acd/pull/9#discussion_r1040876647

Reply:
I think the suggestion was more to remove this check and let `n_acd_bpf_map_add()` return success when the map is negative. In this way, we keep tracking the `probe->acd->n_bpf_map` value correctly; then, if the map can be created later, it will be re-initialized with the right size in `n_acd_ensure_bpf_map_space()`?

Reply:
Yes